synapse/tests/test_lib_storm.py

```diff
@@ -23,6 +23,72 @@ async def test_lib_storm_basics(self):
             msgs = await core.stormlist('$foo=bar $lib.print($foo) if $node { $foo=$node.value() }')
             self.stormIsInPrint('bar', msgs)
 
+            # test storm background command
+            await core.nodes('''
+                $lib.queue.add(foo)
+                [inet:ipv4=1.2.3.4]
+                background {
+                    [it:dev:str=haha]
+                    fini{
+                        $lib.queue.get(foo).put(hehe)
+                    }
+                }''')
+            self.eq((0, 'hehe'), await core.callStorm('return($lib.queue.get(foo).get())'))
+
+            with self.raises(s_exc.StormRuntimeError):
+                await core.nodes('[ ou:org=*] $text = $node.repr() | background $text')
+
+            with self.raises(s_exc.NoSuchVar):
+                await core.nodes('background { $lib.print($foo) }')
+
+            # test the parallel command
+            nodes = await core.nodes('parallel --size 4 { [ ou:org=* ] }')
+            self.len(4, nodes)
+
+            # check that subquery validation happens
+            with self.raises(s_exc.NoSuchVar):
+                await core.nodes('parallel --size 4 { [ ou:org=$foo ] }')
+
+            # check that an exception on inbound percolates correctly
+            with self.raises(s_exc.BadTypeValu):
+                await core.nodes('[ ou:org=* ou:org=foo ] | parallel { [:name=bar] }')
+
+            # check that an exception in the parallel pipeline percolates correctly
+            with self.raises(s_exc.BadTypeValu):
+                await core.nodes('parallel { [ou:org=foo] }')
+
+            nodes = await core.nodes('ou:org | parallel {[ :name=foo ]}')
+            self.true(all([n.get('name') == 'foo' for n in nodes]))
```

Comment:
this is confusing since previous discussion about parallel required readonly to be available?